sunpy/image/transform.py

```diff
@@ -55,32 +60,206 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
     -----
     This algorithm uses an affine transformation as opposed to a polynomial
     geometrical transformation, which by default is `skimage.transform.warp`.
-    One can specify using `scipy.ndimage.affine_transform` as
-    an alternative affine transformation. The two transformations use different
-    algorithms and thus do not give identical output.
+    One can specify using `scipy.ndimage.interpolation.affine_transform`  as an alternative
+    affine transformation. The two transformations use different algorithms
+    and thus do not give identical output.
 
-    When using `skimage.transform.warp` with order >= 4 or using
-    `scipy.ndimage.affine_transform` at all, "NaN" values will be replaced with
-    zero prior to rotation. No attempt is made to retain the "NaN" values.
+    When using for `skimage.transform.warp` with order >= 4 or using
+    `scipy.ndimage.interpolation.affine_transform` at all, "NaN" values will
+    replaced with zero prior to rotation. No attempt is made to retain the NaN
+    values.
 
     Input arrays with integer data are cast to float 64 and can be re-cast using
     `numpy.ndarray.astype` if desired.
 
-    In the case of `skimage.transform.warp`, the image is normalized to [0, 1]
-    before passing it to the function. It is later rescaled back to the original range.
-
     Although this function is analogous to the IDL's ``rot`` function, it does not
     use the same algorithm as the IDL ``rot`` function.
     IDL's ``rot`` calls the `POLY_2D <https://www.harrisgeospatial.com/docs/poly_2d.html>`__
     method to calculate the inverse mapping of original to target pixel
     coordinates. This is a polynomial geometrical transformation.
     Then optionally it uses a bicubic convolution interpolation
     algorithm to map the original to target pixel values.
+
+    If a custom transform function is passed to `method`, it must have the function signature:
+    `foo(image,rmatrix,order,scale,missing,image_center,recenter)`
+    (identical to the parent affine_transform, without the `method` argument),
+    and it must return the new rotated, scaled, translated image.
```

Comment:
My largest concern about this approach is encapsulated in these two sentences. Arguably however, it's a problem which already exists.

It's very easy for us to specify what the arguments of the function must be, takes these arguments etc. It's a lot harder however for us to describe the desired (default) behaviour of the function, such that it would be comparable to the default implementation. This probably isn't something we need to worry about in this PR however, although at somepoint we really should test out the properties of our transform methods.

Reply:
Yeah, I'm not sure the best way to describe the desired result of `rotate`. The chief problem I had developing the openCV equivalent was that the library handles rotation, translation, and scaling operations in a different order and a different coordinate system than skimage/scipy. Even skimage and scipy have different systems between one another! 

Ultimately, we can't control how external libraries implement their manip routines. I think all we can do is to say that the image is rotated by `angle` degrees counterclockwise, translated to the image center (or input argument), and scaled by the `scale` factor; the user would have to do the legwork to verify that their external library operation gives a similar result as the current implementation.

We could make that legwork easier by defining a comparison metric for testing. I have attempted to do this via the symmetric mean absolute percentage error (as seen in the CV example). Otherwise, I'm perfectly happy to boot this to future developers 😄

Reply:
I think having a more detailed description of the algorithm in the docs and then maybe a gallery example describing how to do some comparisons would be great. Otherwise, yeah I think we should punt this to a new issue or something.

Reply:
Ok, I've taken a stab at the description of the algorithm in the docstring, and added a gallery example with map comparisons (in addition to the current one demonstrating a drop-in rotation function). Let me know how these could be improved!